compiler/src/dotty/tools/io/FileWriters.scala

```diff
@@ -25,13 +24,107 @@ import java.util.zip.CRC32
 import java.util.zip.Deflater
 import java.util.zip.ZipEntry
 import java.util.zip.ZipOutputStream
-import scala.language.unsafeNulls
+import scala.collection.mutable
+
+import dotty.tools.dotc.core.Contexts, Contexts.Context
+import dotty.tools.dotc.core.Decorators.em
+
+import dotty.tools.dotc.util.{SourcePosition, NoSourcePosition}
+
+import dotty.tools.dotc.reporting.Message
+import dotty.tools.dotc.report
+
+import dotty.tools.backend.jvm.PostProcessorFrontendAccess.BackendReporting
+import scala.annotation.constructorOnly
 
 /** Copied from `dotty.tools.backend.jvm.ClassfileWriters` but no `PostProcessorFrontendAccess` needed */
 object FileWriters {
   type InternalName = String
   type NullableFile =  AbstractFile | Null
 
+  inline def ctx(using ReadOnlyContext): ReadOnlyContext = summon[ReadOnlyContext]
+
+  sealed trait DelayedReporting {
```

Comment:
Reportor？

Reply:
I reused the name from the code copied from `PostProcessorFrontendAccess` where it was called `BackendReporting`, if some other significant comment comes up ill consider renaming